ui/src/components/settings/sub_pages/profile/mod.rs

```diff
@@ -0,0 +1,182 @@
+    let username = use_state(&cx, || "username".to_owned());
+    let status_message = use_state(&cx, || "status message".to_owned());
+
+
+    let change_banner_text = get_local_text("settings-profile.change-banner");
+    let change_avatar_text = get_local_text("settings-profile.change-avatar");
+
+    let show_texts = !**edit_mode;
+    let show_edit_fields = **edit_mode;
+
+    cx.render(rsx!(
+        div {
+            id: "settings-profile",
+            div {
+                class: "profile-header",
+                div { 
+                    class: "profile-banner", 
+                    img {
+                        class: "profile-banner-photo",
+                        src: "{banner_state}",
+                        height: "100%",
+                        width: "100%",
+                        onclick: move |_| change_profile_image(banner_state),
+                    },
+                    p {class: "change-banner-text", "{change_banner_text}" },
+            },
+                div {
+                    class: "profile-picture",
+                    img {
+                        class: "profile-avatar",
+                        src: "{image_state}",
+                        onclick: move |_| change_profile_image(image_state),
+                    },
+                    p {class: "change-avatar-text", "{change_avatar_text}" },
+                }
+                div {
+                    class: "plus-button", 
+                    Button {
+                        icon: kit::icons::Icon::Plus,
+                        onpress: move |_| change_profile_image(image_state),
+                    },
+                },
+                show_texts.then(|| rsx!(
+                    div {
+                        class: "edit-button", 
+                        Button {
+                            text: get_local_text("settings-profile.edit-button"),
+                            onpress: move |_| edit_mode.set(!edit_mode),
+                        },
+                    },
+                    p { 
+                        class: "username",
+                        "{username}"
+                    },
+                    p { 
+                        class: "status-message",
+                        "{status_message}"
+                    }
+                ))
+                show_edit_fields.then(|| 
+                    {
+                    let new_username_val = use_ref(&cx, String::new);
+                    let new_status_message_val = use_ref(&cx, || format!("{}", status_message));
+                    rsx!(
+                    div {
+                        class: "edit-button", 
+                        Button {
+                            text: get_local_text("settings-profile.save-button"),
+                            onpress: move |_| {
+                                let new_username = new_username_val.with(|i| i.clone());
+                                if !new_username.is_empty() && new_username.len() > 3 {
```

Comment:
```suggestion
                                if !new_username.is_empty() && new_username.len() > 3 && new_username.len() < 32 {
```

Reply:
Could probably do an additional check to return an error if it exceeds the max length.

Reply:
Instead of doing this, you can just use validation on the input element. See the friends page for an example.

Reply:
` // Set up validation options for the input field
    let validation_options = Validation {
        // The input should have a maximum length of 32
        max_length: Some(32),
        // The input should have a minimum length of 4
        min_length: Some(4),
        // The input should only contain alphanumeric characters
        alpha_numeric_only: true,
        // The input should not contain any whitespace
        no_whitespace: true,
    };

    // Set up options for the input field
    let input_options = Options {
        // Enable validation for the input field with the specified options
        with_validation: Some(validation_options),
        // Do not replace spaces with underscores
        replace_spaces_underscore: false,
        // Show a clear button inside the input field
        with_clear_btn: true,
        // Use the default options for the remaining fields
        ..Options::default()
    };

...

                Input {
                    placeholder: get_local_text("friends.placeholder"),
                    icon: Icon::MagnifyingGlass,
                    options: input_options
                },
`